src/main/java/net/revelc/code/formatter/xml/XMLFormatter.java

```diff
@@ -13,19 +13,54 @@
  */
 package net.revelc.code.formatter.xml;
 
+import java.util.Map;
+
+import net.revelc.code.formatter.AbstractCacheableFormatter;
+import net.revelc.code.formatter.ConfigurationSource;
 import net.revelc.code.formatter.Formatter;
 import net.revelc.code.formatter.LineEnding;
-import net.revelc.code.formatter.jsoup.JsoupBasedFormatter;
 
 /**
  * @author yoshiman
- *
+ * @author jam01
  */
-public class XMLFormatter extends JsoupBasedFormatter implements Formatter {
+public class XMLFormatter extends AbstractCacheableFormatter implements Formatter {
+    private XmlDocumentFormatter formatter;
+
+    @Override
+    public void init(Map<String, String> options, ConfigurationSource cfg) {
+        super.initCfg(cfg);
+
+        FormattingPreferences prefs = new FormattingPreferences();
+        String maxLineLength = options.get("maxLineLength");
+        String wrapLongLines = options.get("wrapLongLines");
+        String tabInsteadOfSpaces = options.get("tabInsteadOfSpaces");
+        String tabWidth = options.get("tabWidth");
+        String splitMultiAttrs = options.get("splitMultiAttrs");
+
+        prefs.setMaxLineLength(maxLineLength != null ? Integer.valueOf(maxLineLength) : null);
+        prefs.setTabWidth(tabWidth != null ? Integer.valueOf(tabWidth) : null);
+        prefs.setWrapLongLines(wrapLongLines != null ? Boolean.valueOf(wrapLongLines) : null);
+        prefs.setTabInsteadOfSpaces(tabInsteadOfSpaces != null ? Boolean.valueOf(tabInsteadOfSpaces) : null);
+        prefs.setSplitMultiAttrs(splitMultiAttrs != null ? Boolean.valueOf(splitMultiAttrs) : null);
+
+        this.formatter = new XmlDocumentFormatter(options.getOrDefault("lineending", System.lineSeparator()), prefs);
```

Comment:
@jam01 Is 'lineending' case sensitive here and possible needs to be 'lineEnding'?

Is it also possible to put order of strings to match order of the preference loading?  That makes review a bit easier :)

Reply:
It's 'lineending' because it's supposed to be like json's formatter which configures it when instantiating the formatter, but your comment uncovered that it wasn't setup correctly from the mojo, so thanks.

Solved the order of the strings :+1:

Reply:
Perfect!  Thanks.